### ISSUE TITLE
consider not flagging members in private classes/mixins/extensions/enums w/ type_annotate_public_apis

### ISSUE THREAD

**Issue by pq** [dart-lang/linter#1689]
For example:

```dart
class _Shhh {
  whoops() { // LINT
  }
}
```
lints when it probably shouldn't.

Follow-up from #1688.
